### PR TITLE
start bgp_eoiu_mark service to populate bgp eoiu marker flags for warm start,  if configured so

### PR DESCRIPTION
**Pull request by heidinet2007**
This is to address review comments for
https://github.com/Azure/sonic-buildimage/pull/2823

Other related PR:

https://github.com/Azure/sonic-utilities/pull/655
https://github.com/Azure/sonic-buildimage/pull/2823
https://github.com/Azure/sonic-swss-common/pull/273
https://github.com/Azure/sonic-swss/pull/856

### CODE REVIEW

File: dockers/docker-fpm-frr/start.sh [hunk 1]
```diff
@@ -31,6 +31,11 @@ rm -f /var/run/rsyslogd.pid
 
 supervisorctl start rsyslogd
 
+# start eoiu pulling, only if configured so
+if [ $(sonic-cfggen -d --print-data | grep 'bgp_eoiu' | grep -c 'true') -ge 1 ]; then
```
Comment (pavel-shirshov) [inline dockers/docker-fpm-frr/start.sh]: `sonic-cfggen -d -v bgp_eoiu` could work better here

Comment (heidinet2007) [reply dockers/docker-fpm-frr/start.sh]: Hi Pavel, thanks for your review. However,  "sonic-cfggen -d -v" doesn't seem to do the desired check:

admin@ASW-7004:~$ sonic-cfggen -d --print-data | grep 'bgp_eoiu'
            "bgp_eoiu": "true",   <============ show the  config
admin@ASW-7004:~$ sonic-cfggen -d -v bgp_eoiu
                                               <============ show nothing

admin@ASW-7004:~$ sonic-cfggen -d -v 'WARM_RESTART' 
{'bgp': {'bgp_eoiu': 'true', 'bgp_timer': '120'}}
admin@ASW-7004:~$

Comment (heidinet2007) [reply dockers/docker-fpm-frr/start.sh]: I just updated the diff. Please review it. thanks

Comment (pavel-shirshov) [reply dockers/docker-fpm-frr/start.sh]:
```
if [ $(sonic-cfggen -d -v 'WARM_RESTART.bgp.bgp_eoiu' | grep -c 'true')
```
to
```
if [ $(sonic-cfggen -d -v 'WARM_RESTART.bgp.bgp_eoiu' = 'true' ];
```
or
```
if [[ $(sonic-cfggen -d -v 'WARM_RESTART.bgp.bgp_eoiu' == 'true' ]];
```

File: dockers/docker-fpm-frr/start.sh [hunk 2]
```diff
@@ -33,6 +33,13 @@ rm -f /var/run/rsyslogd.pid
 
 supervisorctl start rsyslogd
 
+# start eoiu pulling, only if configured so
+if [[ $(sonic-cfggen -d -v 'WARM_RESTART.bgp.bgp_eoiu') == 'true' ]]; then
+		supervisorctl start bgp_eoiu_marker
+fi
+
+supervisorctl start bgpcfgd
```
Comment (prsunny) [inline dockers/docker-fpm-frr/start.sh]: Looks like this is added accidently? It was moved to end as part of  https://github.com/Azure/sonic-buildimage/pull/3627

Comment (pavel-shirshov) [reply dockers/docker-fpm-frr/start.sh]: https://github.com/Azure/sonic-buildimage/pull/3752

**Pull request merged**